api/server/server.go

```diff
@@ -42,9 +50,5 @@ func (tapi ToDDApi) Start(cfg config.Config) {
 	serve_url := fmt.Sprintf("%s:%s", tapi.cfg.API.Host, tapi.cfg.API.Port)
 
 	log.Infof("Serving ToDD Server API at: %s\n", serve_url)
-	err := http.ListenAndServe(serve_url, nil)
-	if err != nil {
-		log.Error("Error starting API")
-		os.Exit(1)
-	}
+	return http.ListenAndServe(serve_url, nil)
```

Comment:
Along the same lines of https://github.com/Mierdin/todd/pull/2/files#r57105122 - I think it's useful to create our own error here, since there's context here that will be list if we simply pass the error up the chain.

I do agree with the removal of ```os.Exit(1)``` as we've discussed, provided that the main function detects the error and will quit the ToDD server if ```err != nil```